servers/zts/src/main/java/com/yahoo/athenz/zts/ZTSImpl.java

```diff
@@ -3117,7 +3129,7 @@ X509CertRecord getValidatedX509CertRecord(ResourceContext ctx, final String prov
 
             if (cert.getNotBefore().getTime() < x509CertRefreshResetTime) {
                 x509CertRecord = insertX509CertRecord(ctx, principalName, provider, instanceId,
-                        cert.getSerialNumber().toString(), false);
+                        cert.getSerialNumber().toString(), false, expiryTime, null);
```

Comment:
this one is the recovery mode certificate record. so the correct value should be the expiry of the cert  certificate and not the expiryTime. So we can extract the expiry from `cert` object passed in and use that.